File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
def solution(dartResult):
    answer = 0
    
    stack = []
    
    for idx, item in enumerate(dartResult):
        if item.isdigit():
            if item == "0" and len(stack) >= 1 and stack[-1] == 1:
                stack.pop()
                stack.append(10)
            else:
                stack.append(int(item))
        else:
            if item == "D":
                stack[-1] = stack[-1]**2
            elif item == "T":
                stack[-1] = stack[-1]**3
            elif item == "*":
                stack[-1] *= 2
                if len(stack) >= 2:
                    stack[-2] *= 2
            elif item == "#":
                stack[-1] *= -1
    
    return sum(stack)

File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.js
function solution(dartResult) {
    let answer = [];
    for (const item of dartResult) {
        if (item === "S") {
            continue;
        } else if (item === "D") {
            answer[answer.length - 1] = answer[answer.length - 1]**(2);
        } else if (item === "T") {
            answer[answer.length - 1] = answer[answer.length - 1]**(3);
        } else if (item === "*") {
            answer.forEach((item, index) => {
                if (index === answer.length - 1 || index === answer.length - 2 ) {
                    answer[index] = item * 2;
                }
            });
        } else if (item === "#") {
            answer[answer.length - 1] = answer[answer.length - 1] * (-1);
        } else if (item === "0") {
            if (answer[answer.length - 1] === 1) {
                answer[answer.length - 1] = answer[answer.length - 1] + 9;
            } else {
                answer.push(Number(item));
            }
        }
        else {
            answer.push(Number(item));
        }
    }
    return answer.reduce((sum, num) => sum + num);
}